app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -95,75 +61,28 @@ private void setupItemListener(QuerySnapshot querySnapshots, FirebaseFirestoreEx
             for (QueryDocumentSnapshot doc: querySnapshots) {
                 Map<String, Object> data = new HashMap<>(doc.getData());
                 itemList.add(new Item(doc.getId(), data));
-                itemAdapter.notifyDataSetChanged();
+                unselectAllItems();
             }
         }
     }
 
     /**
-     * Displays the filter menu with options to select the appropriate filter
-     * @param view The view to set the filter menu on
+     * This method unselects all the items in the list displayed and unchecks the corresponding
+     * checkboxes as well.
+     * @see Item
+     * @see ItemAdapter
      */
-    private void showFilterMenu(View view) {
-        PopupMenu popupMenu = new PopupMenu(requireContext(), view);
-        MenuInflater inflater = popupMenu.getMenuInflater();
-        inflater.inflate(R.menu.filter, popupMenu.getMenu());
-
-        popupMenu.setOnMenuItemClickListener(item -> {
-            int itemId = item.getItemId();
-            if (itemId == R.id.filter_by_dates) {
-                View dateFilterView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter_by_dates, null);
-                DateFilterFragment dateFilterFragment = new DateFilterFragment("Modify Date Filter", dateFilterView, this);
-                dateFilterFragment.show(requireActivity().getSupportFragmentManager(), "dates_filter_dialog");
-            } else if (itemId == R.id.filter_by_make) {
-                View makeFilterView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter_by_make, null);
-                MakeFilterFragment makeFilterFragment = new MakeFilterFragment("Modify Make Filter", makeFilterView, this);
-                makeFilterFragment.show(requireActivity().getSupportFragmentManager(), "make_filter_dialog");
-            } else if (itemId == R.id.filter_by_keywords) {
-                View keywordFilterView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter_by_keywords, null);
-                KeywordFilterFragment keywordFilterFragment = new KeywordFilterFragment("Modify Keyword Filter", keywordFilterView, this);
-                keywordFilterFragment.show(requireActivity().getSupportFragmentManager(), "keywords_filter_dialog");
-            } else if (itemId == R.id.filter_by_tags) {
-                View tagFilterView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter_by_tags, null);
-                TagFilterFragment tagFilterFragment = new TagFilterFragment("Modify Tag Filter", tagFilterView, this);
-                tagFilterFragment.show(requireActivity().getSupportFragmentManager(), "tags_filter_dialog");
-            } else {
-                return false;
+    protected void unselectAllItems() {
```

Comment:
This can probably move into Home Select State no?

Reply:
Yeah, can move this, needed it here before since both HomeSelect and HomeBase were using it, but after refactoring my code, I think I can remove unselectAllItems() from the listener, meaning the method will now only be used in HomeSelect